File: IT/2/Input-Output/Esercizi/C++/esercizio13.cpp
/*
file esercizio13.cpp
compilare con: g++ -std=c++11 esercizio13.cpp
*/

#include <iostream>
#include <fstream>

using std::cin;
using std::cout;
using std::endl;
using std::string;
using std::fstream;

int main()
{
	cout << "Scrivi una serie di parole. Invia la stringa vuota per uscire." << endl;

	cout << "File su cui scrivere: ";

	string filename;
	getline(cin, filename);
	fstream doc(filename);

	if (doc) {
		cout << "Il file esiste. Le modifiche saranno aggiunte al file esistente." << endl;
		doc.close();
		doc.open(filename, fstream::out | fstream::app);
	}
	else {
		doc.open(filename, fstream::out);
	}

	while (true) {
		cout << "Scrivi una parola: ";

		string s;
		getline(cin, s);

		if (s.empty()) {
			break;
		}

		doc << s << endl;
	}

	cout << "File \"" << filename << "\" salvato su disco." << endl;
}

File: IT/2/Input-Output/Esercizi/Java/Esercizio13.java
/* file Esercizio13.java */

import java.io.Console;
import java.io.File;
import java.io.FileWriter;
import java.io.BufferedWriter;
import java.io.IOException;

public class Esercizio13 {
	public static void main(String [] args) {
		System.out.println("Scrivi una serie di parole. Invia la stringa vuota per uscire.");

		Console console = System.console();
		String filename = console.readLine("File su cui scrivere: ");

		try {
			File doc = new File(filename);
			FileWriter writer = null;

			if (doc.exists()) {
				System.out.println("Il file esiste. Le modifiche saranno aggiunte al file esistente.");
				writer = new FileWriter(doc, true);
			}
			else {
				writer = new FileWriter(doc);
			}

			BufferedWriter buffer = new BufferedWriter(writer);

			while (true) {
				String s = console.readLine("Scrivi una parola: ");

				if (s.isEmpty()) {
					break;
				}

				buffer.write(s);
				buffer.write(System.lineSeparator());
			}

			buffer.close();

			System.out.println("File \"" + doc.getName() + "\" salvato su disco.");
		}
		catch (IOException e) {
			System.out.println(e);
		}
	}
}

